app/lib/three_scale/reform.rb

```diff
@@ -24,19 +24,15 @@ def save!(*)
     protected
 
     def merge_errors
-      definitions = property_definitions.values
-      definitions.each do |definition|
+      schema.each do |value|
```

Comment:
I found this change pretty tricky, because I don't really understand the documentation of the reform gem very well (not sure if it's mine, or the documentation's fault).

This seems to do the trick, although:

I am not sure if the behavior is 100% equivalent
The code says The #errors method will be removed in Reform 3.0 core. def errors(*args) https://github.com/trailblazer/reform/blob/v2.6.2/lib/reform/contract/validate.rb#L14

Also, there are just three forms, it seems, that use reform,and honestly, I think it might be possible to get rid of it at some point and simplify the implementation.

I think for the current purposes this rework should be enough, but probably we can refactor/remove this later.

Reply:
Which was the problem here? How can I verify it works the same way?

Reply:
The problem was that after the upgrade `self.class.representer_class.representable_attrs[:definitions]` did not exist anymore. The internal data structures in the library changed a lot, and the field definitions are now stored in the `schema`.

I'll try to come of with a short snippet that shows the behavior.

Reply:
@jlledom So, these are the forms that use Reform:

- Notification Preferences - here I don't even know how it uses it, because it doesn't use the Reform's API (`property :name` etc.).
- the other two belong to Onboarding wizard

https://github.com/search?q=repo%3A3scale%2Fporta+include+ThreeScale%3A%3AReform&type=code

From what I can see the errors are not even visualized in the UI, so this `merge_errors` method seems to be pretty useless to me.

However, I added a couple of tests [here](https://github.com/3scale/porta/pull/3725/commits/e49dbcc463a07fb0426baf0798147f4c3d0282a9) to prove that the method works (without `merge_errors` the test fail), and I also confirmed via manual testing that the behavior is the same as in master. Probably except I am converting the field name to symbol for consistency with [ActiveModel::Errors](https://api.rubyonrails.org/classes/ActiveModel/Errors.html), but again, we're not relying on this anywhere, from what I can tell.